tests/evaluate/test_evaluation.py

```diff
@@ -361,6 +363,73 @@ def baseline_model_uri(request):
     return None
 
 
+def test_mlflow_evaluate_logs_traces():
```

Comment:
1. All models types should be supported. Anything other than langchain would just trace the input/output. In langchain, it would create a span for each individual section in langchain.
2. The traces would appear in the traces section as if you were to log a regular trace 
3. I think there isn't a difference between a pyfunc model or function here, because `mlflow.evaluate` does the neccessary steps to turn it all into a `model` with a `predict` method. Here we would be tracking the input/output of this predict method.

Reply:
> Anything other than langchain would just trace the input/output

It's up to the user to decide what level of granularity to use when tracing; if the user instruments each subroutine in their model with @mlflow.trace, then they won't just get overall input / output. We're not limiting / restricting this granularity in this PR, right @jessechancy ?

> The traces would appear in the traces section as if you were to log a regular trace

The key point is that we set the run ID as a tag on the trace, which appears in the UI.

> I think there isn't a difference between a pyfunc model or function here, because mlflow.evaluate does the neccessary steps to turn it all into a model with a predict method. Here we would be tracking the input/output of this predict method.

Same as above - the user might instrument at a level that's more granular than just input / output. Can we confirm this works?